### ISSUE TITLE
a costumize evaluation metric problem in R

### ISSUE THREAD

**Issue by kaleidoscopical** [dmlc/xgboost#223]
I have written a costumize evaluation metric in R as below. However, while I try to train the model using command:  
bst <- xgb.train(param, dtrain, num_round, watchlist, objective = "binary:logistic", fscore)
The R just broke down with the information given below.
I am wondering why this problem occurs and how I can avoid it.
I am a beginner in R, it would be helpful if anyone could kindly give me some advices.
Thank you very much.

####################################################################

*** caught segfault ***
address (nil), cause 'memory not mapped'

Traceback:
 1: .Call("XGBoosterBoostOneIter_R", booster, dtrain, gpair$grad,     gpair$hess, PACKAGE = "xgboost")
 2: xgb.iter.boost(booster, dtrain, gpair)
 3: xgb.iter.update(bst$handle, dtrain, i - 1, obj)
 4: xgb.train(param, dtrain, num_round, watchlist, objective = "binary:logitraw",     fscore)

Possible actions:
1: abort (with core dump, if enabled)
2: normal R exit
3: exit R without saving workspace
4: exit R saving workspace

####################################################################

fscore <- function(preds, dtrain) {
     labels <- getinfo(dtrain, "label")
     preds = preds >= sort(preds,decreasing=TRUE)[49]
     mat = table(preds,labels)
     precision = mat[4]/(mat[4]+mat[3])
     recall = mat[4]/(mat[4]+mat[2])
     f = 2*precision*recall/(precision+recall)
     return(list(metric = "f-score", value = f))
}

**Comment by tqchen** [comment]
can you provide a complete code to reproduce the problem? Would be great if it uses synthetic data or sample data provided by xgboost customized evaluation example

**Comment by tqchen** [comment]
since there was no active updates on this issue, I am closing it for now, please reopen if needed
